### DISCUSSION TITLE
How do I customize Metrics？

### DISCUSSION THREAD

**Discussion by houbinglei** [apache/skywalking#9719] [Q&A]
Endpoint_200 = from(Endpoint.*).filter(responseCode like "5%").percent()；
i used this metric on skywaking version 9.1.0； but i received an err；
i want to caculate the ratio of 5xx for all request;
how do i write the metric expression?

**Comment by wu-sheng** [comment]
Read doc, https://skywalking.apache.org/docs/main/v9.2.0/en/concepts-and-designs/scope-definitions/

responseCode is an int, you can't use `like`. Also this field is going to be replaced in 9.3.0

**Comment by wu-sheng** [comment]
Also, we even provided very clear examples

https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/oal/#examples

Always read the whole docs first.

**Reply by houbinglei** [reply]
thanks very much, i hace read the thole docs, and i found this:
endpoint_abnormal = from(Endpoint.*).filter(httpResponseStatusCode in [404, 500, 503]).count();
but i want to caculate 5xx,  does this version 0f 9.1.0 dont support like 5xx? 
and how can i get the percent of 5xx ?

**Reply by wu-sheng** [reply]
Again and again, it is an int.

**Reply by houbinglei** [reply]
哦～ 嘿嘿， thank you！ i understand～

**Reply by houbinglei** [reply]
excuse me, i have a new question...
i have used metric as this:
endpoint_200 = from(Endpoint.*).filter(httpResponseStatusCode in [200]).count();
but the value for metric endpoint_200 is zero; 
it is abnormal, the 200 request should not be zero?
![image](https://user-images.githubusercontent.com/37292926/193172950-d3fdb92f-a21c-4964-b294-b53dd250d558.png)

**Reply by wu-sheng** [reply]
Have you ever tried to check agent-reported data? response code was not reported before if it is normal(2xx).

Check the data, and check the change log, you may need the latest Java agent release.

**Comment by houbinglei** [comment]
thanks very much! 
now i have used a new metrics: 
service_5xx_percent = from(Service.*).filter(httpResponseStatusCode >=500).filter(httpResponseStatusCode < 600).percent();
but but i received an err again...
could you help me ...

**Reply by wu-sheng** [reply]
Why not check the docs again? The example is provided for percent.

**Reply by houbinglei** [reply]
endpoint_success = from(Endpoint.*).filter(status == true).percent() 
i see this , what is the meaning for  "status == true" ?     i want to caculate the 5xx request percent...
